### PR TITLE
Payroll and payslip page

### PR DESCRIPTION
**Pull request by kaedm**
1. Payroll page, displays the payrolls of each group, you can also run payroll here
1Screenshot_2024-04-17_15-02-31

2. Run payroll modal in the payroll page, Here you can select the Paygroup, Runtype, Start date, End date and Check date of the payroll you want to create
11Screenshot_2024-04-17_15-03-09

3. Payslip page, upon clicking the specific item in the Payroll, it will be redirected to its designated Payslip
2Screenshot_2024-04-17_15-03-47

### CODE REVIEW

**Review by torodeveloper — CHANGES_REQUESTED** [review]
The Payslips table is overcrowded. It's hard to read the data in the cells. Can you display only the primary details? Then display the full details in a separate page or modal when clicking on a row.

**Review by torodeveloper — CHANGES_REQUESTED** [review]
image

It displays `NaN` after clicking on the next button. It should be disabled if I'm already on the last page.

**Comment by torodeveloper** [comment]
I noticed the buttons and navigation on your app have a different appearance compared to Alaina's app. Can someone of you create an app theme first then PR it separately?

**Review by torodeveloper — CHANGES_REQUESTED** [review]
The top navigation links are not working.

**Comment by kaedm** [comment]
1. update on table, only important items are show
image

2. selected row data is now in modal 

image

**Comment by kaedm** [comment]
1. Only important details are now displayed in the payslips page
image

2. upon clicking the row, modal will open and display all the information of that specific row.
image

3. Navbar payroll link is now routed to the payroll page, the other links are static given that thay are not existing in this project. Waiting for themes made by alaina before applying.
image

**Comment by kaedm** [comment]
1. Currently working on the deduction page
image

2. Made the Create deduction modal 
image

**Comment by torodeveloper** [comment]
Can you rebase this to the latest master branch and resolve the conflicts?

File: pages/deduction.json
Comment (torodeveloper) [inline pages/deduction.json:1]: This page should not be included in this PR.

File: components/create-deduction/index.json
Comment (torodeveloper) [inline components/create-deduction/index.json:1]: This modal should not be included in this PR.

File: components/navigation/index.json
Comment (torodeveloper) [inline components/navigation/index.json:1]: We already have a navigation-bar component, do we still need this one?

**Review by torodeveloper — CHANGES_REQUESTED** [review]

**Comment by kaedm** [comment]
1. Pagination on deduction page
image

2. Pagination on Payroll and Payslips page
image

3. Pagination demo 
final

File: stylesheets/style.css [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    align-items: center;
+}
+
+.eHeader div span{
+    height: 100%;
+    display: flex;
+    align-items: center;
+}
+
+.input-content{
```
Comment (torodeveloper) [inline stylesheets/style.css]: Need to format it correctly:

```suggestion
.data-container {
    display: flex;
    width: 100%;
    justify-content:space-between;
}

.data-card {
    background-color: #DEF2F1;
    margin: 1rem;
    padding: .5rem 2rem;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    gap: .5rem;
}   

.table-container {
    position: relative;
}

.spinner {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
}

table {
    height: 100%;
}

.eHeader div {
    width: 100%;
    display: flex;
    justify-content: space-between;
}

.payslip-container {
    display: flex;
}

.payslip-container ul {
    display: flex;
    flex-direction: column;
}

.eHeader div {
    height: 100%;
    display: flex;
    align-items: center;
}

.eHeader div span {
    height: 100%;
    display: flex;
    align-items: center;
}

.input-content {
```

File: app.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
-  "name": "payroll-system",
+  "name": "payroll-system_1",
```
Comment (torodeveloper) [inline app.json]:
```suggestion
  "name": "payroll-system",
```

**Review by torodeveloper — CHANGES_REQUESTED** [review]

File: stylesheets/style.css (hunk 1 above)
Comment (kaedm) [reply stylesheets/style.css]: .data-container{
    display: flex;
    width: 100%;
    justify-content:space-between;
}

.data-card{
    background-color: #DEF2F1;
    margin: 1rem;
    padding: .5rem 2rem;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    gap: .5rem;
}   

.table-container{
    position: relative;
}

.spinner {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
}

table{
    height: 100%;
}

.eHeader div{
    width: 100%;
    display: flex;
    justify-content: space-between;
}

.payslip-container{
    display: flex;
}

.payslip-container ul{
    display: flex;
    flex-direction: column;
}

.eHeader div{
    height: 100%;
    display: flex;
    align-items: center;
}

.eHeader div span{
    height: 100%;
    display: flex;
    align-items: center;
}

.input-content{
    display: flex;
    flex-direction: column;
} 
is this correct?

**Review by torodeveloper — DISMISSED** [review]

**Review by torodeveloper — APPROVED** [review]

**Pull request merged**